pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -261,8 +265,11 @@ func (c *StaticResourceController) AddKubeInformers(kubeInformersByNamespace v1h
 	return ret
 }
 
-func (c *StaticResourceController) AddInformer(informer cache.SharedIndexInformer) *StaticResourceController {
-	c.factory.WithInformers(informer)
+func (c *StaticResourceController) AddInformer(set sets.Set[cache.SharedIndexInformer], informer cache.SharedIndexInformer) *StaticResourceController {
+	if !set.Has(informer) {
```

Comment:
I think it would make sense to fix all controllers in the same way, not just `StaticResourceController`. There is a place where controller `Factory` adds event handlers: https://github.com/openshift/library-go/blob/ac3ba9eb16a23890629f719f3fe11428ad395796/pkg/controller/factory/factory.go#L298-L304

IMO it could skip `informer` + `f.informers[i].filter` tuple that was already added.